aws-lc-rs/src/ec/key_pair.rs

```diff
@@ -224,3 +293,57 @@ fn compute_ecdsa_signature<'a>(
 
     Ok(&mut signature[0..out_sig_len])
 }
+
+/// Elliptic curve private key.
+pub struct PrivateKey<'a>(&'a EcdsaKeyPair);
+
+impl Debug for PrivateKey<'_> {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        f.write_str(&format!("EcdsaPrivateKey({:?})", self.0.algorithm.id))
+    }
+}
+
+/// Elliptic curve private key data encoded as a big-endian fixed-length integer.
+pub struct EcPrivateKeyBin {
+    _priv: (),
+}
+
+/// Elliptic curve private key as a DER-encoded `ECPrivateKey` (RFC 5915) structure.
+pub struct EcPrivateKeyRfc5915Der {
+    _priv: (),
+}
+
+impl PrivateKey<'_> {
+    /// Exposes the private key encoded as a big-endian fixed-length integer.
+    ///
+    /// For most use-cases, `EcdsaKeyPair::to_pkcs8()` should be preferred.
+    ///
+    /// # Errors
+    /// `error::Unspecified` if serialization failed.
+    pub fn to_bin(&self) -> Result<Buffer<'static, EcPrivateKeyBin>, Unspecified> {
+        unsafe {
+            let buffer = ec::marshal_private_key_to_buffer(
+                self.0.algorithm.id,
+                &self.0.evp_pkey.as_const(),
+            )?;
+            Ok(Buffer::<EcPrivateKeyBin>::new(buffer))
+        }
+    }
+
+    /// Serializes the key as a DER-encoded `ECPrivateKey` (RFC 5915) structure.
+    ///
+    /// # Errors
+    /// `error::Unspecified`  if serialization failed.
+    pub fn to_der(&self) -> Result<Buffer<'static, EcPrivateKeyRfc5915Der>, Unspecified> {
```

Comment:
Which library and/or language is serializing to `ECPrivateKey` directly that we need to inter-opt with?

Reply:
One popular library is BC-FIPS (in Java).